signer-service/src/api/services/pendulum.service.js

```diff
@@ -56,7 +62,32 @@ exports.sendStatusWithPk = async () => {
   const { fundingAccountKeypair, fundingAmountRaw } = getFundingData(apiData.ss58Format, apiData.decimals);
   const { data: balance } = await apiData.api.query.system.account(fundingAccountKeypair.address);
 
-  if (Big(balance.free.toString()).gte(fundingAmountRaw)) {
+  const tokensToCheck = Object.keys(TOKEN_CONFIG);
+  let isTokensSufficient = true;
+
+  // Wait for all required token balances check.
+  await Promise.all(
+    tokensToCheck.map(async (token) => {
+      const tokenConfig = TOKEN_CONFIG[token];
+      console.log(`Checking token ${token} balance...`);
+      const tokenBalanceResponse = await apiData.api.query.tokens.accounts(
+        fundingAccountKeypair.address,
+        tokenConfig.pendulumCurrencyId,
+      );
+
+      const tokenBalance = Big(tokenBalanceResponse?.free?.toString() ?? '0');
+      const maximumSubsidyAmountRaw = Big(tokenConfig.maximumSubsidyAmountRaw);
+      const subsidyRatio = tokenBalance.div(maximumSubsidyAmountRaw);
+
+      if (subsidyRatio.lt(SUBSIDY_MINIMUM_RATIO_FUND_UNITS)) {
```

Comment:
I've chosen a ratio between the balance and the maximum subsidy amount (defined to be `10` for now). This was mostly to avoid adding decimals to `TOKEN_CONFIG`.